extensions/script-libraries/mime/mime.livecodescript

```diff
@@ -0,0 +1,1271 @@
+
+Description:
+Use <MIMEEncode> to encode data for transfer
+
+Returns:
+The encoded content
+
+References:
+textEncode (function)
+
+*/
+
+function MIMEEncode pEncoding, pData, pTextEncoding
+   if pTextEncoding is not empty then
+      put textEncode(pData, pTextEncoding) into pData
+   end if
+   
+   switch pEncoding
+      case "quoted-printable"
+         return __EncodeQuotedPrintable(pData) for value
+         break
+      case "base64"
+         return __EncodeBase64(pData) for value
+         break
+   end switch
+   
+   return "unsupported encoding" for error
+   
+end MIMEEncode
+
+private function __EncodeBase64 pData
+   return replaceText(base64Encode(pData), return, crlf)
+end __EncodeBase64
+
+private function __EncodeQuotedPrintable pData
+   local tData
+   local tLineLength = 1
+   local tNumBytes
+   put the number of bytes of pData into tNumBytes
+   local tIndex = 1
+   repeat for each byte tByte in pData
+      if tLineLength is 76 then
+         put "=" & crlf after tData
+         put 1 into tLineLength
+      end if
+      
+      local tByteNum
+      put byteToNum(tByte) into tByteNum
+      
+      local tPrevByteNum
+      
+      switch
+         case tByteNum is 10 and tPrevByteNum is not 13
+            put crlf after tData
+            put 1 into tLineLength
+            break 
+         case (tByteNum is 9 or tByteNum is 32) and  tLineLength is not 76 and tIndex is not tNumBytes
+         case tByteNum >= 33 and tByteNum <= 126 and tByteNum is not 61
+            put tByte after tData
+            add 1 to tLineLength
+            break
+         default
+            if tLineLength >= 74 and \
+                  not (tLineLength is 74 and tIndex is tNumBytes) then
+               put "=" & crlf after tData
+               put 3 into tLineLength
+            end if
+            
+            local tConverted
+            put baseConvert(tByteNum,10,16) into tConverted
+            if the length of tConverted is 1 then
+               put 0 before tConverted
+            end if
+            put "=" & tConverted after tData
+            add 3 to tLineLength
+            break
+      end switch
+      
+      put tByteNum into tPrevByteNum
+      add 1 to tIndex
+   end repeat
+   return tData
+end __EncodeQuotedPrintable
+
+/**
+
+Build a multipart mime content type from an array of pre-encoded content
+
+Description:
+
+Parameters:
+pType (enum): The type of multipart content:
+
+- "mixed": for a generic mixed set of parts
+- "alternative": for representing the same data in multiple formats
+- "parallel": for parts intended to be viewed simultaneously
+- "digest": for multipart entities in which each part has a default type of "message/rfc822"
+
+pContent (array):
+A numerically indexed array of pre-encoded parts including headers. If no headers are present
+a leading crlf should be present.
+
+Returns:
+The multipart encoded content
+
+*/
+
+function MIMEMultipart pType, pContent
```

Comment:
I think it might be slightly more ergonomic to make this:

```
function MIMEMultipart pContent, pType
```

and allow `pType` to default to `mixed` if not specified.

Please name this function with a verb.  What would you call a function that parses a multipart MIME blob?